libkbfs/folder_branch_ops.go

```diff
@@ -1911,7 +1987,7 @@ func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
 		} else {
 			bid = fbo.bid
 		}
-		err := mdops.PutUnmerged(ctx, md, bid)
+		mdID, err = fbo.doPutWithCancellationCheck(ctx, lState, md, bid)
```

Comment:
I'm talking about the `if errPut == context.Canceled {` case above in `finalizeMDWriteLocked`.  If we hit that case, then the grace period is already over.  But you're already put a lot of effort into possibly returning a `nil` error for that case, which could in theory be discarded by future checks of `ctx.Done()`.